### ISSUE TITLE
Displaying :: and [] with the same color as constructors

### ISSUE THREAD

**Issue by ysalmon** [ocaml/tuareg#201]
Recent tuareg version allow displaying of constructors like ``None`` with a specific color.

However, this specific color is not used for list predefined constructors ``::`` and ``[]``, which surprised some of my students (at least, they understood that these are constructors !).

And indeed, formally, they are right.

However I do not think this should be extended to ``[a; b; …; z]`` as this is just syntactic sugar for a constructor.

**Comment by ysalmon** [comment]
@Chris00 it is not possible for me to report issues in https://github.com/Chris00/tuareg ; the relevant tab does not appear.

**Comment by Chris00** [comment]
Right, you have to do it in this repo.

**Comment by ysalmon** [comment]
Hello, I finally had time to check on this. Coloration of :: and [] works fine but there are differences in parenthesis, pipe and operator color. I do not know if this is related or intended.

Here a comparison between the git version and the regular 2.2.0.

![Screenshot_20190709_091847](https://user-images.githubusercontent.com/536850/60867254-b1fd3b80-a22a-11e9-94a7-e36071a56e44.png)
![Screenshot_20190709_091809](https://user-images.githubusercontent.com/536850/60867255-b1fd3b80-a22a-11e9-8d22-d31d8d2d1776.png)

**Comment by Chris00** [comment]
Yes, this is on purpose.  Coloring braces and pipes is costly (there is another—closed—issue about this) and does not improve readability IMHO.  Coloring braces was mostly for optional arguments in function definitions but now the same effect is achieved by other (more efficient) means.  `+` is not colored because it is costly (and too many colors do no help) but if you want so, set `tuareg-highlight-all-operators` to `t` in your init file.
